colin-api/src/colin_api/models/business.py

```diff
@@ -671,3 +671,93 @@ def reset_corp_states(cls, cursor, event_ids: List):
+                    left join filing on event.event_id = filing.event_id and filing.filing_typ_cd in ('OTANN', 'ANNBC')
+                where corp.CORP_NUM=:corp_num
+                order by filing.period_end_dt desc nulls last
+                """,
+                corp_num=corp_num
+            )
+            business = cursor.fetchone()
+            if not business:
+                raise BusinessNotFoundException(corp_num=corp_num)
+
+            # add column names to resultset to build out correct json structure and make manipulation below more robust
+            # (better than column numbers)
+            business = dict(zip([x[0].lower() for x in cursor.description], business))
+            # get all assumed, numbered/corporation, translation names
+            corp_names = CorpName.get_current(cursor=cursor, corp_num=corp_num)
+            assumed_name = None
+            corp_name = None
+            for name_obj in corp_names:
+                if name_obj.type_code == CorpName.TypeCodes.ASSUMED.value:
+                    assumed_name = name_obj.corp_name
+                    break
+                elif name_obj.type_code in [CorpName.TypeCodes.CORP.value, CorpName.TypeCodes.NUMBERED_CORP.value]:
+                    corp_name = name_obj.corp_name
+
+            # get last ledger date from EVENT table and add to business record
+            # note - FILE event type is correct for new filings; CONVOTHER is for events/filings pulled over from COBRS
+            cursor.execute(
+                """
+                select max(EVENT_TIMESTMP) from EVENT
+                where EVENT_TYP_CD in ('FILE', 'CONVOTHER') and CORP_NUM=:corp_num
+                """,
+                corp_num=corp_num
+            )
+            last_ledger_timestamp = cursor.fetchone()[0]
+            business['last_ledger_timestamp'] = last_ledger_timestamp
+            # if this is an XPRO, get correct jurisdiction; otherwise, it's BC
+            if business['corp_typ_cd'] == 'XCP':
+                business['jurisdiction'] = business['can_jur_typ_cd']
+                if business['can_jur_typ_cd'] == 'OT':
+                    business['jurisdiction'] = business['othr_juris_desc']
+            else:
+                business['jurisdiction'] = 'BC'
+
+            # convert to Business object
+            business_obj = Business()
+            business_obj.business_number = business['bn_15']
+            business_obj.corp_name = assumed_name if assumed_name else corp_name
+            business_obj.corp_num = business['corp_num']
+            business_obj.corp_state = business['corp_state']
+            business_obj.corp_state_class = business['corp_state_class']
+            business_obj.corp_type = business['corp_typ_cd']
+            business_obj.email = business['admin_email']
+            business_obj.founding_date = convert_to_json_datetime(business['recognition_dts'])
+            business_obj.jurisdiction = business['jurisdiction']
```

Comment:
Add a field `homeIdentifier` to the business object that gets populated with `home_jurisdiction_num`